core/src/ir/implementation.rs

```diff
@@ -68,14 +68,40 @@ impl TryFrom<ItemImpl> for Implementation {
     }
 }
 
+impl Implementation {
+    /// Maps the dependencies in the method signatures
+    pub fn dependencies(&self) -> Vec<Type> {
+        let mut deps: Vec<Type> = vec![];
+        for item in &self.items {
+            if let ImplementationItem::Method(method) = item {
+                method.inputs.clone().into_iter().for_each(|parameter| {
+                    if !deps.iter().any(|typ| typ == &parameter.type_) {
+                        deps.push(parameter.type_);
```

Comment:
Don't we also need to check against `Self`? It isn't guaranteed that the type will be transformed when this function is called

Reply:
You mean `Self` or `self`?